sdk/storage/azure-storage-blob/azure/storage/blob/_models.py

```diff
@@ -739,13 +739,13 @@ class BlobBlock(DictMixin):
 
     :param str block_id:
         Block id.
-    :param str state:
-        Block state. Possible values: committed|uncommitted
+    :param Enum state:
+        Block state. Possible values: BlockState.COMMITTED | BlockState.UNCOMMITTED
     """
     block_id: str
-    state: Union[str, Enum]
+    state: Enum
 
-    def __init__(self, block_id: str, state: Union[str, Enum] = BlockState.Latest) -> None:
+    def __init__(self, block_id: str, state: Enum = BlockState.LATEST) -> None:
```

Comment:
So, this type isn't quite right. It should be `state: BlockState` not just generic `Enum`. `BlockState` inherits from `str` now so I don't think its breaking to longer accept a string?